bb8/src/inner.rs

```diff
@@ -174,7 +176,14 @@ where
         let start = Instant::now();
         let mut delay = Duration::from_secs(0);
         loop {
-            match shared.manager.connect().await {
+            let conn = match shared.manager.connect().await {
+                Err(e) => Err(e),
+                Ok(mut c) => match self.on_acquire_connection(&mut c).await {
+                    Err(e) => Err(e),
+                    Ok(_) => Ok(c),
+                },
+            };
```

Comment:
The repeated `Err(e) => Err(e)` arms can probably be factored out by using some combinators, can you look into that? And while you're in there, please add an empty line between first `let conn = match` and `match conn {` blocks?

Reply:
Updated. Looks a bit more compact.